frontend/src/employee-frontend/components/employee/EmployeePinCodePage.tsx

```diff
@@ -2,31 +2,38 @@
 //
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 
 import { Container, ContentArea } from 'lib-components/layout/Container'
 import { Gap } from 'lib-components/white-space'
 import Title from 'lib-components/atoms/Title'
 import { Label, P } from 'lib-components/typography'
-import InputField from 'lib-components/atoms/form/InputField'
+import InputField, { InputInfo } from 'lib-components/atoms/form/InputField'
 import { FixedSpaceColumn } from 'lib-components/layout/flex-helpers'
 import { faLockAlt } from 'lib-icons'
 
 import { useTranslation } from 'employee-frontend/state/i18n'
-import { updatePinCode } from 'employee-frontend/api/employees'
+import { isPinCodeLocked, updatePinCode } from 'employee-frontend/api/employees'
 import AsyncButton from 'lib-components/atoms/buttons/AsyncButton'
 import Button from 'lib-components/atoms/buttons/Button'
+import { Result } from '../../../lib-common/api'
+import { AlertBox } from '../../../lib-components/molecules/MessageBoxes'
 
 export default React.memo(function EmployeePinCodePage() {
   const { i18n } = useTranslation()
   const [pin, setPin] = useState<string>('')
   const [error, setError] = useState<boolean>(false)
+  const [pinLocked, setPinLocked] = useState<Result<boolean>>()
+
+  useEffect(() => {
+    void isPinCodeLocked().then(setPinLocked)
+  }, [setPinLocked])
 
   function isValidNumber(pin: string) {
     const int = parseInt(pin)
-    const str = int.toString()
-    return str.length === 4
+    const str = int.toString().padStart(4, '0')
+    return pin == str && pin.length === 4
```

Comment:
It might just be my preference for regex but this could be slightly more readable/robust:

```ts
function isValidNumber(pin: string) {
  return /^\d{4}$/.test(pin)
}
```

That would be exactly 4 characters, all of which must be explicitly digits 0-9.